### DISCUSSION TITLE
forward wireguard server connection to wireguard client?

### DISCUSSION THREAD

**Discussion by RobotsAreCrazy** [qdm12/gluetun#1335] [Q&A]
Hi,

is it possible, let's say you have a wireguard gluetun client setup....could a wireguard server be setup to allow a user to connect and then have their traffic routed through the client tunnel?

often wondered could this be done?

Thanks

**Comment by bnhf** [comment]
@RobotsAreCrazy 

This is basically the same as #1323. Technically, this should be possible, especially if everything was being done at the host OS level. Trying to do it with Docker containers, is trickier since best performance is achieved by using the kernel implementation of WireGuard (or OpenVPN), rather than a container implementation. Gluetun achieves this through the NET_ADMIN directive on compatible kernels.

However, there's a straightforward way to do this by exposing the built-in Gluetun Shadowsocks server port to the Internet, and do the first leg using a Shadowsocks client which has encryption and password protection. The second leg would be over the WireGuard VPN.

A single container doing both jobs, with relatively easy setup -- especially if you've used Gluetun before.
